lib/stator/alias.rb

```diff
@@ -2,77 +2,99 @@
 
 module Stator
   class Alias
+    attr_reader :machine, :name, :namespace, :attr_name, :states, :not, :opposite_args, :constant, :scope
+
     def initialize(machine, name, options = {})
       @machine    = machine
       @name       = name
-      @namespace  = @machine.namespace
-      @full_name  = [@namespace, @name].compact.join('_')
+      @namespace  = machine.namespace
       @states     = []
       @not        = false
       @opposite   = nil
       @constant   = options[:constant]
       @scope      = options[:scope]
     end
 
+    def attr_name
+      @attr_name ||= generate_attr_name
+    end
+
     def is(*args)
-      @states |= args.map(&:to_s)
+      @states |= args.map(&:to_sym)
     end
 
     def is_not(*args)
       @not = true
       is(*args)
     end
 
+    alias not? not
+
     def opposite(*args)
-      @opposite = args
+      # set the incoming args for opposite as opposite
+      @opposite_args = args
     end
 
     def evaluate
       generate_methods
 
-      if @opposite
-        op = @machine.state_alias(*@opposite)
+      return if opposite_args.blank?
 
-        op.is(*@states)     if @not
-        op.is_not(*@states) unless @not
-      end
+      # this will generate the alias for the opposite
+      op = machine.state_alias(*opposite_args)
+
+      op.is(*states)     if not?
+      op.is_not(*states) unless not?
     end
 
-    protected
+    private
+
+    def current_states
```

Comment:
Doesn't feel like `current_states` is the right name. This is representing the inverse states, correct?

Reply:
agreed, inverse_states feels nice

Reply:
Resolved in ba07cdeb